systemtest/src/test/java/io/strimzi/systemtest/security/PodSecurityProfilesST.java

```diff
@@ -60,169 +68,158 @@ public class PodSecurityProfilesST extends AbstractST {
+            .editSpec()
+                .withClassName("org.apache.kafka.connect.file.FileStreamSinkConnector")
+                .addToConfig("topics", testStorage.getTopicName())
+                .addToConfig("file", Constants.DEFAULT_SINK_FILE_PATH)
+                .addToConfig("key.converter", "org.apache.kafka.connect.storage.StringConverter")
+                .addToConfig("value.converter", "org.apache.kafka.connect.storage.StringConverter")
+            .endSpec()
+            .build());
+
+        // Messages produced to Main Kafka Cluster (source) will be sinked to file, and mirrored into targeted Kafkas to later verify Operands work correctly.
+        LOGGER.info("Deploy producer: {} and produce {} messages into Kafka: {} in Ns: ", testStorage.getProducerName(), testStorage.getClusterName(), messageCount);
         final KafkaClients kafkaClients = new KafkaClientsBuilder()
             .withTopicName(testStorage.getTopicName())
-            .withMessageCount(testStorage.getMessageCount())
+            .withMessageCount(messageCount)
             .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
             .withProducerName(testStorage.getProducerName())
             .withConsumerName(testStorage.getConsumerName())
             .withNamespaceName(testStorage.getNamespaceName())
             .withUsername(testStorage.getUsername())
             .withPodSecurityPolicy(PodSecurityProfile.RESTRICTED)
             .build();
-
         resourceManager.createResourceWithWait(extensionContext, kafkaClients.producerStrimzi());
         ClientUtils.waitForProducerClientSuccess(testStorage);
 
-        resourceManager.createResourceWithWait(extensionContext, KafkaMirrorMakerTemplates.kafkaMirrorMaker(testStorage.getClusterName(), testStorage.getTargetClusterName(), testStorage.getClusterName(), ClientUtils.generateRandomConsumerGroup(), 1, false).build());
-
-        final List<Pod> kafkaClusterAndKafkaMirrorMakerPods = PodUtils.getKafkaClusterPods(testStorage);
-        // add KafkaMirrorMaker2 Pod
-        kafkaClusterAndKafkaMirrorMakerPods.addAll(kubeClient().listPods(testStorage.getNamespaceName(), testStorage.getClusterName(), Labels.STRIMZI_KIND_LABEL, KafkaMirrorMaker2.RESOURCE_KIND));
-
-        // 1. check the generated structure of SecurityContext of Kafka and KafkaMirrorMaker2 Pods
-        // verifies that (i.) Pods and (ii.) Containers has proper generated SC
-        verifyPodAndContainerSecurityContext(kafkaClusterAndKafkaMirrorMakerPods);
-
-        // 2. check that KMM2 mirrors messages
-        resourceManager.createResourceWithWait(extensionContext, kafkaClients.consumerStrimzi());
+        // verifies that (i.) Pods and (ii.) Containers have proper generated SC
+        final List<Pod> kafkaClusterAndKafkaBridgePods = PodUtils.getKafkaClusterPods(testStorage);
```

Comment:
This is not used anywhere right? Shouldn't it be used in the `verifyPodAndContainerSecurityContext();` on line 181?

Also, I think it's possible that we can collect all "Strimzi pods" using one call, I can check it and help you with that.

Reply:
great catch i actually passed wrong variable.